distributed/comm/ucx.py

```diff
@@ -103,14 +102,55 @@ def peer_address(self):
         return self._peer_addr
 
     async def write(self, msg, serializers=None, on_error="message"):
-        nbytes = sys.getsizeof(msg)
-        await self.ep.send_msg(msg, nbytes)
-        return nbytes
+        frames = await to_frames(msg,
+                                 serializers=serializers,
+                                 on_error=on_error)  # TODO: context=
+
+        msg = b''.join(frames)
+        # nbytes = sys.getsizeof(msg)
+
+        # Do we need any of the `stuck.pack` stuff? Is that just
+        # padding the messages for TCP?
+        # I've included it now for unpacking.
+        lengths = [nbytes(frame) for frame in frames]
+        length_bytes = ([struct.pack('Q', len(frames))] +
+                        [struct.pack('Q', x) for x in lengths])
+        b = b''.join(length_bytes + frames)  # small enough, send in one go
+
+        # TODO: exception handling
+        length_bytes = sys.getsizeof(b)
+        await self.ep.send_msg(b, length_bytes)
+        return length_bytes
```

Comment:
So you shouldn't need to do much of this.  Ideally it looks a bit more like the following:

```python
frames = await to_frames(...)
await self.ep.send_msg(struct.pack('Q', len(frames)))  # send number of frames
for frame in frames:
    await self.ep.send_msg(frame)
```

UCX will handle the other logic around the number of bytes in each frame for us.  It's a bit smarter than Tornado here.